### ISSUE TITLE
Aura smart thermostat support

### ISSUE THREAD

**Issue by ian682** [webdjoe/pyvesync#208]
Any chance to support the aura smart thermostat, LTM-A401S-WUS? I picked two of them up on prime day and didn't realize the vesync HA integration didn't support them.

**Comment by ian682** [comment]
If I need to capture anything, let me know and how you need it done.

**Comment by webdjoe** [comment]
Currently, the only way to capture is to use an Android Emulator and following steps 4 & 5 [here](https://redhuntlabs.com/wp-content/uploads/2021/10/Ultimate-Guide-to-SSL-Pinning-Bypass-RedHunt-Labs-Attack-Surface-Management.pdf) and then using an SSL proxy capturing app, such as charles proxy  or Burp Suite. [This](https://medium.com/@roy2477/decrypting-android-app-ssl-traffic-c480e2143743) is a good guide to setting up an Android emulator. and ssl proxy.

**Comment by ian682** [comment]
I managed to get the captures and emailed them to you. let me know if you need anything else.

**Comment by TheDarkPreacher** [comment]
anything new going on with this? Just got an Aura 400S thermostat and would like to control it with HA but the Vesync integration doesn't pick it up.

**Comment by ian682** [comment]
I haven't heard anything. Honestly starting to think about moving to a different thermostat since this one isn't in home assistant. Google Assistant keeps dumping features I use and I'm not happy about it, lol. The only thing I have now that isn't supported in home assistant is two of these thermostats.

**Comment by webdjoe** [comment]
Sorry for delay, life has been busy. Do you still need this to integrate?

**Comment by TheDarkPreacher** [comment]
I certainly do. The wife has cut me off from buying a different smart thermostat until the Aura 400S dies... and that thing will probably outlast the house, since Levoit makes good stuff.

**Comment by mpunjack** [comment]
> Sorry for delay, life has been busy. Do you still need this to integrate?
@webdjoe  were you able to get this to integrate?

**Comment by TheDarkPreacher** [comment]
Just circling back around to check on the status of this. Any news?

**Comment by webdjoe** [comment]
Yes, please test out PR #310 on branch https://github.com/webdjoe/pyvesync/tree/device-structure

This is will the new library structure that is async - read the README of the new branch - https://github.com/webdjoe/pyvesync/tree/device-structure

I didn't have time to document the thermostat methods, so please refer to https://github.com/webdjoe/pyvesync/blob/device-structure/src/pyvesync/devices/vesyncthermostat.py

https://github.com/webdjoe/pyvesync/blob/device-structure/src/pyvesync/base_devices/thermostat_base.py

for methods. The state details are in the `device.state` attribute.  The thermostats can be directly accessed via the `manager.devices.thermostats` or just `manager.devices` if you only have thermostats.

**Comment by cdnninja** [comment]
Could someone with this device contact me on Discord. I am open to adding this code to home assistant now that it exists in the library but would need someone to test.
